esm_analysis/carbon.py

```diff
@@ -256,27 +256,85 @@ def _check_variables(ds):
     return sensitivity
 
 
+@check_xarray(0)
+def co2_flx_ocean_sensitivities(ds):
+    """Compute sensitivity of oceanic co2_flux to changes in driver variables.
+
+    .. math: co2_flx_ocean = co2trans * (co2atm - co2ocean) * (1 - ice)
+
+    Args:
+        ds (xr.Dataset): containing cmorized variables:
+                         * co2_flx_ocean [kg m-2 s-1]: surface air-sea CO2 flux
+                         * co2ocean[ppm CO2]: partial pressure of CO2 in
+                                              surface sea water
+                         * CO2[ppm]: partial pressure of atmospheric CO2 at the
+                                     surface
+                         * co2trans [10-9 mol s m-4] : transfer velocity of
+                                                       ocean/atmosphere CO2 flux
+                         * seaice [%]: fraction of grid cell covered by ice
+
+    Returns:
+        sensitivity (xr.Dataset):
+
+    References:
+        * Lovenduski, Nicole S., Nicolas Gruber, Scott C. Doney, and Ivan D. Lima.
+          “Enhanced CO2 Outgassing in the Southern Ocean from a Positive Phase of
+          the Southern Annular Mode.” Global Biogeochemical Cycles 21, no. 2
+          (2007). https://doi.org/10/fpv2wt.
+        * Sarmiento, Jorge Louis, and Nicolas Gruber. Ocean Biogeochemical Dynamics.
+          Princeton, NJ: Princeton Univ. Press, 2006., p.421, eq. (10:3:1)
+
+    """
+
+    def _check_variables(ds):
+        requiredVars = ['co2_flx_ocean', 'CO2', 'co2ocean', 'co2trans', 'seaice']
+        if not all(i in ds.data_vars for i in requiredVars):
+            missingVars = [i for i in requiredVars if i not in ds.data_vars]
+            raise ValueError(
+                f"""Missing variables needed for calculation:
+            {missingVars}"""
+            )
+
+    _check_variables(ds)
+
+    # Sensitivities are based on the time-mean for each field. This computes
+    # sensitivities at each grid cell.
+    # TODO: Add keyword for sliding mean, as in N year chunks of time to
```

Comment:
Definitely would be a follow up PR, unless we can make the sliding calculation a standardized function. We found out this was pretty important based on my test cases!